doc/how-to/app/app_roles.rst

```diff
@@ -0,0 +1,476 @@
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application_role_cfg/byeer.lua
+    :language: lua
+    :dedent:
+
+A role cannot be started without its dependencies.
+This means that all the dependencies of a role should be defined in the ``roles`` configuration parameter:
+
+..  code-block:: yaml
+
+    instance001:
+      roles: [ greeter, byeer ]
+
+You can find the full example here: `application_role_cfg <https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/config/instances.enabled/application_role_cfg>`_.
+
+
+
+.. _roles_create_custom_role_init:
+
+Adding initialization code
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can add initialization code to a role by defining and calling a function with an arbitrary name at the top level of a module, for example:
+
+..  code-block:: lua
+
+    local function init()
+        -- ... --
+    end
+
+    init()
+
+For example, you can :ref:`create spaces <box_schema-space_create>`, define :ref:`indexes <concepts-data_model_indexes>`, or :ref:`grant privileges <configuration_credentials>` to specific users or roles.
+
+See also: :ref:`roles_create_space`.
+
+
+
+.. _roles_create_space:
+
+Specifics of creating spaces
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create a space in a role, you need to make sure that the target instance is in read-write mode (its :ref:`box.info.ro <box_introspection-box_info>` is ``false``).
+You can check an instance state by subscribing to the ``box.status`` event using :ref:`box.watch() <box-watch>`:
+
+..  code-block:: lua
+
+    box.watch('box.status', function()
+        -- creating a space
+        -- ...
+    end)
+
+
+..  NOTE::
+
+    Given that a role may be enabled when an instance is already in read-write mode,
+    you also need to execute schema initialization code from :ref:`apply() <roles_create_custom_role_apply>`.
+    To make sure a space is created only once, use the :ref:`if_not_exists <space_opts_if_not_exists>` option.
+
+
+
+.. _roles_life_cycle:
+
+Roles life cycle
+----------------
+
+A role’s life cycle includes the stages described below.
+
+..  _roles_life_cycle_loading_roles:
+
+1)  *Loading roles*
+
+    On each run, all roles are loaded in the order they are specified in the :ref:`configuration <roles_create_custom_role_config>`.
+    This stage takes effect when a role is enabled or an instance with this role is restarted.
+    At this stage, a role executes the :ref:`initialization code <roles_create_custom_role_init>`.
+
+    A role cannot be started if it has :ref:`dependencies <roles_create_custom_role_dependencies>` that are not specified in a configuration.
+
+    ..  NOTE::
+
+        Dependencies do not affect the order in which roles are loaded.
+        However, the ``validate()``, ``apply()``, and ``stop()`` functions are executed taking dependencies into account.
+        Learn more in :ref:`roles_life_cycle_dependencies_specifics`.
+
+
+..  _roles_life_cycle_stopping_roles:
+
+2)  *Stopping roles*
```

Comment:
Maybe `Spotting removed roles` or smth like that? the current wording surprised me and took some time to understand which roles we're talking about